src/store.ts

```diff
@@ -99,9 +100,8 @@ export class Store<
   /**
    * Read-only property that holds references to the store state.
    */
-  public get state(): TRootState {
-    return this.store.state;
-  }
+
+  public readonly state: Readonly<TState<TOptions['state']>>;
```

Comment:
This change works, but I suspect that it breaks Store#replaceState. However, there is an alternative constructor that lets users pass in a Vuex store as a parameter. If they want to replace the state, then they can use that function. I'm okay with sacrificing that functionality to get types out of the store's state.